### PR TITLE
Student B - Display Comments pop up with selected item's details

### LINKED ISSUE #2: [3pt]  - Add tests for comments counter - Student B
Make sure that the function for the comments counter is covered with unit tests using Jest.

Make sure that you have tests for all edge cases.

### LINKED ISSUE #5: [3pt]  - Add comments counter - Student B
Even if the API gives you the number of comments you will create a specific function to calculate it.

Make sure that the correct number is displayed on the Comments pop up.

Respect the following rules:

- Each counter should be implemented as a separate module.
- A counter function should look for specific DOM elements (e.g. for the comments counter it should look for comments) and make the counting based on what is actually displayed on the page.
- A counter function should cover all the edge cases you can think about.

### LINKED ISSUE #8: [4pt]  - Create feature: add new comment - Student B
Implement the "add new comment" form on the Comments pop up according to the wireframe:

Screenshot 2021-06-27 at 22 32 36

When the user clicks on the "Comment" button, the data is recorded in the [Involvement API](https://www.notion.so/microverse/Involvement-API-869e60b5ad104603aa6db59e08150270) and the screen is updated.

### LINKED ISSUE #11: [3pt]  - Display comments for a given item on the Comments pop-up - Student B
When the popup loads, the webapp retrieves data from the [Involvement API](https://www.notion.so/microverse/Involvement-API-869e60b5ad104603aa6db59e08150270) to show the item comments.

Screenshot 2021-06-27 at 22 21 00

This task **does not** include the counter of the comments.

### LINKED ISSUE #14: [4pt]  -Display Comments pop up with selected item's details - Student B
Create a comments popup only with the top section (displaying details of the selected item)

Screenshot 2021-06-27 at 22 10 45

When the popup loads, the webapp retrieves data from the selected API and shows details about the selected item.

Add event to button prepared by your teammate that will open your popup. You need to communicate with your team member to make sure that you do not block each other (most likely the easiest solution is to merge the list of items feature to the dev branch before you try to add that event).

### LINKED ISSUE #16: [0.5pt] Add footer - group task
Create footer according to the wireframe: 

Screenshot 2021-06-27 at 21 42 29

### LINKED ISSUE #18: [0.5pt] Set up the project with Webpack and Jest- group task
- Set up Gitflow
- Set up webpack
- Set up a JavaScript testing library (Jest)

### LINKED ISSUE #19: [1pt] Choose assets - group task
Choose images, text, icons, fonts.

### LINKED ISSUE #20: [0.5pt] Find external API - group task
Find an API to base the development of the webapp around it. The API should allow you to
  - Get a list of items with a unique item id (or generate the unique id).
  - For a given item, get detailed information about it.
choose an APIs that doesn't need authentication. if you select an API that requires authentication, you should implement it on your own. Also, if you select an API that provides image resources your webapp will be more visual.

### LINKED ISSUE #21: Setup Kanban Board
https://github.com/KIsmoilov/JS-Capstone-Project/projects/1

In our team, there are two team members

Student A = @soesandarwin2201 
Student B = @KIsmoilov

### PR DESCRIPTION
**Pull request by KIsmoilov**
Display Comments pop up with selected item's details

The following features have been implemented:

- Comments pop-up window has been connected with the external API.
- Display Comments pop-up with selected item details. 
- Created a mock comment button to ensure the JS code works dynamically.
- Worked on the styling of the pop-up window.
- To check the dynamic functionality of JS code please change the id of the following element in the index.html
`<div class="meal-img" data-id= "2">`

Thanks for the code review @soesandarwin2201

### CODE REVIEW

File: src/modules/popUpDom.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      </form>`;
+
+  popUpContainer.appendChild(popUp);
+};
+
+const getSelectedMeal = async (selectedIdCat) => {
+  const selectedMeals = await getData();
+  if (selectedIdCat !== undefined) {
+    displaySelectedMeal(selectedMeals.categories.find((meal) => meal.idCategory === selectedIdCat));
+  }
```
Comment (soesandarwin2201) [inline src/modules/popUpDom.js]: Can You explain why you use this method for selectedMeals fucntions.

**Review by soesandarwin2201 — APPROVED** [review]
## STATUS: Approved  💯

## Hi @KIsmoilov 👋, 

Good job so far! You did a great job !  There is nothing else to say other than continue working on project
## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- GitHub flow is followed. :heavy_check_mark:
- PR has a good title and summary. ✔️

68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f6c3366516333614356304871707a5763552f67697068792e676966


_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

File: src/modules/popUpDom.js (hunk 1 above)
Comment (KIsmoilov) [reply src/modules/popUpDom.js]: @soesandarwin2201 I am using this function to retrieve the id of the parent element of the comment button and I am comparing it with the id of the meal category so that I can show the right meal on the pop-up. If there is need for any improvements, please let me know

**Comment by KIsmoilov** [comment]
Display Comments pop up with selected item's details

The following features have been implemented:

- Created the function to display comments for a given item on the Comments pop-up page.
- Created the function to add new comments.
- Created the function to count the number of comments 
- Added tests for comments counter 

Thanks for the code review @soesandarwin2201

File: src/modules/popUpDom.js (hunk 1 above)
Comment (soesandarwin2201) [reply src/modules/popUpDom.js]: @KIsmoilov  Thanks for your explanation , now i can use that method for my like Button function ✨

File: src/modules/popUpDom.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    const commentDiv = document.createElement('div');
+    commentDiv.className = 'comment-item';
+    commentDiv.innerHTML = `
+     <p class="commentDate">${comments[i].creation_date}</p>
+     <p class="commenterName">${comments[i].username}:</p>
+     <p class="commentContent">${comments[i].comment}</p>
+     `;
+    commentMainDiv.appendChild(commentDiv);
+  }
+};
```
Comment (soesandarwin2201) [inline src/modules/popUpDom.js:20]: It would be better if you would more javascript method like (map, filter, forEach,find). In this case you  can use ForEach Method .

**Review by soesandarwin2201 — CHANGES_REQUESTED** [review]
Hi @KIsmoilov , 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

almost4

 ## Highlights 💯 
- Good job following the advice of the previous reviewer ✔️ 
- No linter errors. :heavy_check_mark:
- GitHub flow is followed. :heavy_check_mark:
- PR has a good title and summary. ✔️
- Asynchronous functions have been used ✔️
- Nice design ✔️

 ## Required Changes ♻️
 - please make the mobile design for you popup
 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

File: src/modules/popUpDom.js (hunk 2 above)
Comment (KIsmoilov) [reply src/modules/popUpDom.js:20]: @soesandarwin2201 Thank you so much for your comments. In this place, I had to use indexes so that I couldn`t use other methods.

**Comment by KIsmoilov** [comment]
Display Comments pop up with selected item's details

The following changes have been implemented:

- Used the media queries to make the Comments pop-up window responsive to any screen size.
- Checked the functionality by creating a mock button on Html 

Thanks for the code review @soesandarwin2201

**Review by soesandarwin2201 — APPROVED** [review]
Hi @KIsmoilov , 

Great work on making the changes 👏 
You've done well implementing the requested changes,

almost4

 ## Highlights 💯 
- Good job following the advice of the previous reviewer ✔️ 
- No linter errors. :heavy_check_mark:
- GitHub flow is followed. :heavy_check_mark:
- PR has a good title and summary. ✔️
- Asynchronous functions have been used ✔️
- Nice design ✔️


 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
